internal/webhook/external/quay_test.go

```diff
@@ -0,0 +1,120 @@
+package external
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/indexer"
+)
+
+func TestQuayHandler(t *testing.T) {
+	const testURL = "https://webhooks.kargo.example.com/nonsense"
+
+	const testProjectName = "fake-project"
+
+	testScheme := runtime.NewScheme()
+	require.NoError(t, kargoapi.AddToScheme(testScheme))
+
+	const testToken = "mysupersecrettoken"
+	testSecretData := map[string][]byte{
+		quaySecretDataKey: []byte(testToken),
+	}
+
+	for _, testCase := range []struct {
+		name       string
+		client     client.Client
+		secretData map[string][]byte
+		req        func() *http.Request
+		assertions func(*testing.T, *httptest.ResponseRecorder)
+	}{
+		{
+			name:       "request body too large",
+			secretData: testSecretData,
+			req: func() *http.Request {
+				body := make([]byte, 2<<20+1)
+				req := httptest.NewRequest(
+					http.MethodPost,
+					testURL,
+					io.NopCloser(bytes.NewBuffer(body)),
+				)
+				return req
+			},
+			assertions: func(t *testing.T, rr *httptest.ResponseRecorder) {
+				require.Equal(t, http.StatusRequestEntityTooLarge, rr.Code)
+				res := map[string]string{}
+				err := json.Unmarshal(rr.Body.Bytes(), &res)
+				require.NoError(t, err)
+				require.Contains(t, res["error"], "content exceeds limit")
+			},
+		},
+		{
+			name:       "success -- push event",
```

Comment:
Nit: Not worth qualifying as "push event" when it's the only event type we're responding to.

Separate from that, I'd suggest adding a partial success test case like the one in #4352

```suggestion
			name:       "success",
```